src/utilities.jl

```diff
@@ -45,6 +55,14 @@ Equivalent to `permutedims(x, perm)`.
 """
 swapdims(perm) = Base.Fix2(permutedims, perm)
 
+"""
+    applyactivation(activation, x)
+
+Apply an activation function to a given input.
+Equivalent to `activation.(x)`.
+"""
+applyactivation(activation, x) = activation.(x)
```

Comment:
Might be a good place to consider upstreaming https://github.com/avik-pal/Lux.jl/blob/aafb2b88455661670ed12055365ba5f573ce4219/src/nnlib.jl#L199 to NNlib and the layer https://github.com/avik-pal/Lux.jl/blob/aafb2b88455661670ed12055365ba5f573ce4219/src/layers/basic.jl#L120 to Flux

Reply:
I'm not a fan of needing to wrap activations like that in a layer. NNlib already provides a way for layers to use a fast activation implementation e.g. `tanh_fast`. I don't see why GPU activations can't hook into the same mechanism.

Reply:
Mostly this is because of error messages. With anonymous functions, the error is unclear. While part of Flux's advantage is writing it that way just works, it also means that for newer uses when stuff errors out it is kinda hard to track down

Reply:
I totally agree but I think Julia provides us better options than creating essentially a domain specific language for function calling (which is what this solution is). Perhaps if you think this is nice but not necessary for the PR, we can open an issue on Flux for more discussion? I agree that having a nice error message across all activations would be good.

Also, it seems NNlibCUDA already dispatches to the faster activations without explicitly needing anything from the user. So the performance detail is moot.

Reply:
That makes sense. I will open up an issue